### PR TITLE
Dev / Stage Sync - 062722

### PR DESCRIPTION
**Pull request by Gbarra9**
Merging the following issues from dev: 

Issue #541 - https://github.com/episphere/connect/issues/541

Issue #543 - https://github.com/episphere/connect/issues/543

Issue #544 - https://github.com/episphere/connect/issues/544

Issue #545 - https://github.com/episphere/connect/issues/545

Issue #534 - https://github.com/episphere/connect/issues/534

### CODE REVIEW

File: src/events.js [hunk 1]
```diff
@@ -1919,19 +1940,58 @@ export const goToParticipantSearch = () => {
     document.getElementById('navBarSearch').click();
 }
 
-export const addEventSpecimenLinkForm = (formData) => {
+export const addEventSpecimenLinkForm = (formData, visitConcept) => {
```
Comment (anthonypetersen) [inline src/events.js:1943]: visitConcept not being used, needs to be removed.

File: src/events.js [hunk 2]
```diff
@@ -1919,19 +1940,58 @@ export const goToParticipantSearch = () => {
...
 
     if (document.getElementById('navBarParticipantCheckIn')) document.getElementById('navBarParticipantCheckIn').dataset.connectId = connectId;
 
-    form.addEventListener('submit', e => {
+    form.addEventListener('submit', async (e) => {
         e.preventDefault();
-        btnsClicked(connectId, formData);
+        const query = `connectId=${parseInt(connectId)}`;
+        const participant  = await findParticipant(query);
+        const data = participant.data[0];
```
Comment (anthonypetersen) [inline src/events.js:1953]: can this break if participant is null?

**Review by anthonypetersen — CHANGES_REQUESTED** [review]

File: src/events.js (hunk 1 above)
Comment (geniosys) [reply src/events.js:1943]: It's being used in line 1930

File: src/pages/specimen.js [hunk 3]
```diff
@@ -90,7 +93,7 @@ export const specimenTemplate = async (data) => {
     // addEventBarCodeScanner('scanSpecimenIDBarCodeBtn', 0, masterSpecimenIDRequirement.length);
     // if(document.getElementById('scanAccessionIDBarCodeBtn')) addEventBarCodeScanner('scanAccessionIDBarCodeBtn');
     generateBarCode('connectIdBarCode', data.Connect_ID);
-    addEventSpecimenLinkForm(formData);
+    addEventSpecimenLinkForm(formData, visit.concept);
```
[the remark this answers is not in the document]
Comment (geniosys) [reply src/pages/specimen.js:96]: In events.js, I am declaring function on line #1922 where the second argument is visitConcept.
That visitConcept is being used on line#1930.

While calling the function in speciment.js, I have to provide the second parameter(i.e visitConcept) and thats what I am doing on line 95.

File: src/events.js (hunk 2 above)
Comment (geniosys) [reply src/events.js:1953]: I followed the same logic found in several other instances. See some examples below:
https://github.com/episphere/biospecimen/blob/876dd95a79c57db541d48119d071f3ac92fa2a1a/src/events.js#L2561

https://github.com/episphere/biospecimen/blob/876dd95a79c57db541d48119d071f3ac92fa2a1a/src/events.js#L139

https://github.com/episphere/biospecimen/blob/876dd95a79c57db541d48119d071f3ac92fa2a1a/src/events.js#L1790

https://github.com/episphere/biospecimen/blob/876dd95a79c57db541d48119d071f3ac92fa2a1a/src/events.js#L1820)

Comment (geniosys) [reply src/events.js:1953]: So for example, in the first instance shared above (i.e in function addEventReturnToCollectProcess), data variable is passed in tubeCollectedTemplate and I don't see any null check/validation in that function, so I am assuming that findParticipant won't give null response.

**Review by anthonypetersen — APPROVED** [review]

**Pull request merged**